ext/JumpProcessesKernelAbstractionsExt.jl

```diff
@@ -0,0 +1,178 @@
+module JumpProcessesKernelAbstractionsExt
+
+using JumpProcesses
+using KernelAbstractions, Adapt
+using PoissonRandom, RandomNumbers
+using StaticArrays
+
+function DiffEqBase.solve(jump_prob::JumpProblem, alg::SimpleTauLeaping;
+        backend = CPU(),
+        trajectories = 1,
+        seed = nothing,
+        dt = error("dt is required for SimpleTauLeaping."))
+
+    # boilerplate from SimpleTauLeaping method
+    @assert isempty(jump_prob.jump_callback.continuous_callbacks) # still needs to be a regular jump
+    @assert isempty(jump_prob.jump_callback.discrete_callbacks)
+    prob = jump_prob.prob
+    
+    # Run vectorized solve
+    ts, us = vectorized_solve(jump_prob, SimpleTauLeaping(); backend, trajectories, seed, dt)
+
+    # Convert to CPU for inspection
+    ts = Array(ts)
+    us = Array(us)
+
+    sol = DiffEqBase.build_solution(prob, alg, ts, us,
+        calculate_error = false,
+        interp = DiffEqBase.ConstantInterpolation(ts, us))
+end
+
+# Define an immutable struct to hold trajectory-specific data
+struct TrajectoryData{U <: StaticArray, P, T}
+    u0::U
+    p::P
+    tspan::Tuple{T, T}
+end
+
+# Define an immutable struct to hold common jump data
+struct JumpData{R, C}
+    rate::R
+    c::C
+    numjumps::Int
+end
+
+# SimpleTauLeaping kernel
+@kernel function simple_tau_leaping_kernel(@Const(probs_data), _us, _ts, dt, @Const(rj_data),
+                                          current_u_buf, rate_cache_buf, counts_buf, local_dc_buf,
+                                          seed::UInt64)
+    i = @index(Global, Linear)
+
+    # Get thread-local buffers
+    @inbounds begin
+        current_u = view(current_u_buf, :, i)
+        rate_cache = view(rate_cache_buf, :, i)
+        counts = view(counts_buf, :, i)
+        local_dc = view(local_dc_buf, :, i)
+    end
+
+    # Get problem data
+    @inbounds prob_data = probs_data[i]
+    u0 = prob_data.u0
+    p = prob_data.p
+    tspan = prob_data.tspan
+
+    # Extract jump data
+    rate = rj_data.rate
+    num_jumps = rj_data.numjumps
+    c = rj_data.c
+
+    # Initialize current_u from u0
+    @inbounds for k in 1:length(u0)
+        current_u[k] = u0[k]
+    end
+
+    n = Int((tspan[2] - tspan[1]) / dt) + 1
+    state_dim = length(u0)
+
+    # Get input/output arrays
+    ts_view = @inbounds view(_ts, :, i)
+    us_view = @inbounds view(_us, :, :, i)
+
+    # Initialize first time step and state
+    @inbounds ts_view[1] = tspan[1]
+    @inbounds for k in 1:state_dim
+        us_view[1, k] = current_u[k]
+    end
+
+    # Initialize RNG once per trajectory
+    rng = Xorshifts.Xoroshiro128Plus(seed + i)
+
+    # Main loop
+    for j in 2:n
+        tprev = tspan[1] + (j-2) * dt
+
+        # Compute rates and scale by dt
+        rate(rate_cache, current_u, p, tprev)
+        rate_cache .*= dt
+
+        # Poisson sampling
+        @inbounds for k in 1:num_jumps
+            counts[k] = pois_rand(rng, rate_cache[k])
```

Comment:
https://docs.nvidia.com/cuda/curand/device-api-overview.html

CUDA has a builtin Poisson RNG, wouldn't we expect that to be more appropriate for GPU use?

Reply:
AMD offers it too in ROCRAND: [differentiable programming for odes review](https://rocm.docs.amd.com/projects/rocRAND/en/latest/api-reference/python-api.html#rocrand.PRNG.poisson). Can we not dispatch between NVDIA vs. AMD, and otherwise then fall back to `pois_rand` if in neither of those? I'd imagine there might be substantial performance benefits to using the vendor provided versions.